app/models/ext_management_system.rb

```diff
@@ -622,11 +622,33 @@ def destroy_queue
     task.id
   end
 
+  def ems_workers
+    MiqWorker.find_alive.where(:queue_name => queue_name)
```

Comment:
Was talking about this earlier with @jrafanie, but now I'm confused...

How does this work with workers that list multiple queue names? Will that worker not get caught by this? Is that okay?

Reply:
Good point.  I think this is a bug in the original PR: https://github.com/ManageIQ/manageiq/pull/16755

I noticed this in testing.  It appears that ones such as a refresh worker in amazon which have multiple queue_names for the managers don't get picked up here.  For example:

```
irb(main):007:0> pp MiqWorker.where.not(:queue_name => nil).pluck(:id, :type, :queue_name)
[[32, "MiqReportingWorker", "reporting"],
 [37, "ManageIQ::Providers::Amazon::CloudManager::EventCatcher", "ems_11"],
 [42, "MiqPriorityWorker", "generic"],
 [43, "MiqGenericWorker", "generic"],
 [45, "MiqEventHandler", "ems"],
 [46,
  "ManageIQ::Providers::Amazon::CloudManager::RefreshWorker",
  "[\"ems_11\", \"ems_12\", \"ems_13\"]"]]
```

Instead, once the managers are removed, `sync_workers` realizes these workers are NOT needed anymore and they exit.  Therefore, it's possible the refresh worker for amazon is still running after the manager/ems is deleted.

Reply:
cc @agrare ^ (amazon refresh workers aren't killed here and instead `sync_workers` does it via `stop_worker`...  (I believe this is how it worked before this PR)

Reply:
Additionally, amazon event catcher doesn't seem to respond to `TERM` so it takes 10 minutes before event catchers get terminated in pods.  I need to test this in appliances to confirm how it works there.

Reply:
Well that's unfortunate, for dequeue we had to change the query to `queue_name in (?)` if the queue was an array.

(I'm starting to think the multi-ems-queue_name thing is more trouble than it is worth and we should just delegate queue_name to the parent manager)

Reply:
Ok, so to get back to this PR.  @agrare Where should I open an issue on the multi-ems-queue_name not killing workers before the managers/ems get deleted?  It's amazon here but it sounds like this is a general problem and we can solve that after this PR.

Reply:
@jrafanie right it is not just amazon, can you open it in core?

Reply:
ok, I opened https://github.com/ManageIQ/manageiq/issues/20307 to track that issue.